include/api.php

```diff
@@ -3276,72 +3276,706 @@ function api_oauth_access_token($type) {
+		$height = $ph->getHeight();
+
+		// create a new resource-id if not already provided
+		$hash = ($photo_id == null) ? photo_new_resource() : $photo_id;
+
+		if ($mediatype == "photo") {
+			// upload normal image (scales 0, 1, 2)
+			logger("photo upload: starting new photo upload", LOGGER_DEBUG);
+
+			$r =$ph->store(local_user(), $visitor, $hash, $filename, $album, 0, 0, $allow_cid, $allow_gid, $deny_cid, $deny_gid, $desc);
+			if (! $r)
+				logger("photo upload: image upload with scale 0 (original size) failed");
+
+			if($width > 640 || $height > 640) {
+				$ph->scaleImage(640);
+				$r = $ph->store(local_user(),$visitor, $hash, $filename, $album, 1, 0, $allow_cid, $allow_gid, $deny_cid, $deny_gid, $desc);
+				if (! $r)
+					logger("photo upload: image upload with scale 1 (640x640) failed");
+			}
+
+			if($width > 320 || $height > 320) {
+				$ph->scaleImage(320);
+				$r = $ph->store(local_user(), $visitor, $hash, $filename, $album, 2, 0, $allow_cid, $allow_gid, $deny_cid, $deny_gid, $desc);
+				if (! $r)
+					logger("photo upload: image upload with scale 2 (320x320) failed");
+			}
+			logger("photo upload: new photo upload ended", LOGGER_DEBUG);
+		}
+		elseif ($mediatype == "profileimage") {
+			// upload profile image (scales 4, 5, 6)
+			logger("photo upload: starting new profile image upload", LOGGER_DEBUG);
+
+			if($width > 175 || $height > 175) {
+				$ph->scaleImage(175);
+				$r = $ph->store(local_user(),$visitor, $hash, $filename, $album, 4, $profile, $allow_cid, $allow_gid, $deny_cid, $deny_gid, $desc);
+				if (! $r)
+					logger("photo upload: profile image upload with scale 4 (175x175) failed");
+			}
+
+			if($width > 80 || $height > 80) {
+				$ph->scaleImage(80);
+				$r = $ph->store(local_user(),$visitor, $hash, $filename, $album, 5, $profile, $allow_cid, $allow_gid, $deny_cid, $deny_gid, $desc);
+				if (! $r)
+					logger("photo upload: profile image upload with scale 5 (80x80) failed");
+			}
+
+			if($width > 48 || $height > 48) {
+				$ph->scaleImage(48);
+				$r = $ph->store(local_user(), $visitor, $hash, $filename, $album, 6, $profile, $allow_cid, $allow_gid, $deny_cid, $deny_gid, $desc);
+				if (! $r)
+					logger("photo upload: profile image upload with scale 6 (48x48) failed");
+			}
+			$ph->__destruct();
+			logger("photo upload: new profile image upload ended", LOGGER_DEBUG);
+		}
+
+		if ($r) {
+			// create entry in 'item'-table on new uploads to enable users to comment/like/dislike the photo
+			if ($photo_id == null && $mediatype == "photo")
+				PostPhotoItem($hash, $allow_cid, $deny_cid, $allow_gid, $deny_gid, $filetype, $visibility);
+
+			// on success return image data in json/xml format (like /api/friendica/photo does when no scale is given)
+			return prepare_photo_data($type, false, $hash);
+		}
+		else
+			throw new InternalServerErrorException("image upload failed");
+	}
+
+	function PostPhotoItem($hash, $allow_cid, $deny_cid, $allow_gid, $deny_gid, $filetype, $visibility = false) {
```

Comment:
Standards: Please name this function with the most prevalent convention, which is `lower_case` for global functions.